subworkflows/local/align_merge_long/main.nf

```diff
@@ -0,0 +1,84 @@
+//
+// Align with minimap2 and merge with samtools
+//
+
+include { MINIMAP2_ALIGN           } from '../../../modules/nf-core/minimap2/align/main'
+include { BAM_INDEX_STATS_SAMTOOLS } from '../../local/bam_index_stats_samtools/main'
+include { SAMTOOLS_MERGE           } from '../../../modules/nf-core/samtools/merge/main'
+include { CALCULATE_BASEQUALITY    } from '../../../modules/local/calculate_basequality/main'
+
+workflow ALIGN_MERGE_LONG {
+    take:
+    ch_reads      // channel (optional): [ val(meta), [ path(reads) ] ]
+    ch_alignments // channel (optional): [ val(meta), path(alignments) ]
+    ch_index      // channel (mandatory): [ val(meta2), path(index) ]
+    ch_fasta      // channel (mandatory) : [ val(meta3), path(fasta) ]
+    ch_fai        // channel (mandatory) : [ val(meta4), path(fai) ]
+
+    main:
+    ch_versions = Channel.empty()
+
+    // Map reads with minimap2 - per lane
+    // (nf-core module always sorts BAM output)
+    def mm2_output_bam = true
+    def mm2_custom_bam_index_ext = ''
+    def mm2_use_paf_cigar_fmt = false
+    // work around BAM limit to CIGAR length which can be exceeded with long reads
+    // https://github.com/lh3/minimap2#working-with-65535-cigar-operations
+    def mm2_write_long_cigar_to_bam_tag = true
+    MINIMAP2_ALIGN(
+        ch_reads,
+        ch_index,
+        mm2_output_bam,
+        mm2_custom_bam_index_ext,
+        mm2_use_paf_cigar_fmt,
+        mm2_write_long_cigar_to_bam_tag,
+    )
+
+    ch_versions = ch_versions.mix(MINIMAP2_ALIGN.out.versions.first())
+
+    // Remove laneId, read_group, flowcellId from the metadata to enable sample based grouping
+    ch_bams = MINIMAP2_ALIGN.out.bam
+        .map { meta, bam ->
+            def newMeta = meta.clone()
+            newMeta.remove('laneId')
```

Comment:
do we expect laneID in long reads?

Reply:
It's not forbidden by the metadata schema

Reply:
okay then merging them might also be required